File: results/methods.cs
using System;
using Godot;
using Godot.Collections;


public partial class methods : Godot.Node
{
	public void Empty()
	{
	}

	public void Reassign()
	{
		reassign += 2;
	}

	public void Assign()
	{
		assign = 2;
	}

	//test1
	public double Init(double v = 1.0)
	{//test2
		//test3
		Returning(7.0);
		//test4
	}

	public double Returning(double v)
	{
		Empty();
		return 1.0;
	}

	public void Declare()
	{
		var declare = 2;
	}

	public double ReturnInference(double param = 5.0)
	{
		var val = 2;
		return val * param;
	}

}

File: results/methods.cpp
#include "methods.hpp"
#include <godot_cpp/core/class_db.hpp>

void methods::empty()
{
}

void methods::reassign()
{
	reassign += 2;
}

void methods::assign()
{
	assign = 2;
}

float methods::init(float v)
{//test2
	//test3
	returning(7.0);
	//test4
}

float methods::returning(float v)
{
	empty();
	return 1.0;
}

void methods::declare()
{
	int declare = 2;
}

float methods::return_inference(float param)
{
	int val = 2;
	return val * param;;static void methods::_bind_methods() {
	ClassDB::bind_method(D_METHOD("empty"), &methods::empty);
	ClassDB::bind_method(D_METHOD("reassign"), &methods::reassign);
	ClassDB::bind_method(D_METHOD("assign"), &methods::assign);
	ClassDB::bind_method(D_METHOD("init", "v"), &methods::init);
	ClassDB::bind_method(D_METHOD("returning", "v"), &methods::returning);
	ClassDB::bind_method(D_METHOD("declare"), &methods::declare);
	ClassDB::bind_method(D_METHOD("return_inference", "param"), &methods::return_inference);

}

}

